### DISCUSSION TITLE
Page type differs from type used in the database

### DISCUSSION THREAD

**Discussion by RachelBrodsky** [inducer/relate#881] [Q&A]
Hello!

I am trying to create a flow with both SurveyTextQuestions and SurveyChoiceQuestions, but I get the following error: 
![image](https://user-images.githubusercontent.com/67277602/151625756-a90ae139-ad67-41fd-9f5d-ae7bf62a4c77.png)

Do you know what the problem is/how to resolve this?

Best,
Rachel

**Comment by inducer** [comment]
If you change the question type, you also need to change the ID of the question. (Otherwise, code for one question type would find data belonging to another in the database and would very likely get quite confused.)

**Reply by RachelBrodsky** [reply]
Yes, I understand that. All the question ids in the flow (really in the whole repo) are unique, however there are both survey _text_ question and survey _choice_ questions in the same flow. The error is occurring at the first instance of a survey choice question. Do I still need to change the id in some way?

**Reply by inducer** [reply]
Multiple question types within a flow is totally A-OK.

Your problem (AFAIU) is not about uniqueness within the flow. It's about that question ID having been used with a different type of question *at any point in the past* (from which resulted records in the database). To avoid accidentally associating this old data with a non-matching question type, you need to change the ID of that question.
